app/models/observation.rb

```diff
@@ -193,6 +193,61 @@ class Observation < AbstractModel
   # Automatically (but silently) log destruction.
   self.autolog_events = [:destroyed]
 
+  include ScopesForTimestamps
+
+  # NOTE: Experimental; not tested yet. Needs tests.
+  scope :of_name, lambda {
+    |name,
+     include_synonyms = false,
+     include_subtaxa = false,
+     include_all_name_proposals = false,
+     of_look_alikes = false|
+
+    names = [name]
+    names = name.synonyms if include_synonyms
+    if include_subtaxa
+      names.each do |n|
+        names << Name.subtaxa_of(n)
+      end
+    end
+
+    if include_all_name_proposals
+      joins(:namings).where(namings: { name: names })
+    elsif of_look_alikes
+      joins(:namings).where(namings: { name: names }).
+        where(Observation[:name] != name)
+    else
+      where(name: names)
+    end
+  }
+
+  scope :of_name_like,
+        ->(name) { where(name: Name.text_name_includes(name)) }
+  scope :with_name, -> { where.not(name: Name.unknown) }
+  scope :without_name, -> { where(name: Name.unknown) }
+  scope :by_user, ->(user) { where(user: user) }
+  scope :at_location, ->(location) { where(location: location) }
+  scope :in_region,
+        ->(where) { where(Observation[:where].matches("%#{where}%")) }
+  scope :is_collection_location, -> { where(is_collection_location: true) }
+  scope :not_collection_location, -> { where(is_collection_location: false) }
+  scope :for_project, lambda { |project|
+    joins(:project_observations).
+      where(ProjectObservation[:project_id] = project)
+  }
+  scope :with_image, -> { where.not(thumb_image: nil) }
+  scope :without_image, -> { where(thumb_image: nil) }
+  scope :with_location, -> { where.not(location: nil) }
+  scope :without_location, -> { where(location: nil) }
+  scope :notes_include,
+        ->(notes) { where(Observation[:notes].matches("%#{notes}%")) }
+  scope :with_notes, -> { where.not(notes: Observation.no_notes) }
+  scope :without_notes, -> { where(notes: Observation.no_notes) }
+  scope :with_sequence, -> { where(sequence: true) }
```

Comment:
Is a scope better than
```ruby
joins(:sequences)
```
If so, will this work:
```suggestion
  scope :with_sequence, -> { joins(sequences) }
```

Reply:
Hm. Joins are usually much slower. When I check `Observation.with_sequence.to_sql` in `rails console --sandbox`, i find that it does not execute a join, so i imagine `where(sequence: true)` is probably the quicker way.

Reply:
`sequence` isn't an attribute of `observation`, so, for instance:
```ruby
3.0.4 :002 > Observation.with_sequence.count
   (1.6ms)  SELECT COUNT(*) FROM `observations` WHERE `observations`.`sequence` = TRUE
/home/vagrant/.rvm/gems/ruby-3.0.4/gems/mysql2-0.5.4/lib/mysql2/client.rb:148:in `_query': Mysql2::Error: Unknown column 'observations.sequence' in 'where clause' (ActiveRecord::StatementInvalid)
/home/vagrant/.rvm/gems/ruby-3.0.4/gems/mysql2-0.5.4/lib/mysql2/client.rb:148:in `_query': Unknown column 'observations.sequence' in 'where clause' (Mysql2::Error)
```
Am I misunderstanding this?

Reply:
Nope. Those scopes I wrote are total nonsense! Sorry! Autopilot... I can't blame VS Code either. :)

Do you think the scopes (with join, as you wrote them!) would be useful?
If so, I'll add them for Comments also.